core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -132,7 +132,7 @@ private HTTPClient(
    * shares the same base uri, mapper, and HTTP client as the parent, thus not requiring any
    * additional resource allocation.
    */
-  private HTTPClient(HTTPClient parent, AuthSession authSession) {
+  HTTPClient(HTTPClient parent, AuthSession authSession) {
```

Comment:
Please use the builder of this class instead of changing the visibility of the constructor.

Reply:
In fact, I'll revert this part (the second commit) from this PR because that was a kind of experiment to see how we could test the new interface in HTTPClient. Agreed, with @nastra that this isn't the way we'd want to go.